main.go

```diff
@@ -0,0 +1,209 @@
+func Main(args []string, in io.Reader, out io.Writer) int {
+	plainArgs := args
+	service, err := findServiceType(plainArgs[1])
+	if err != nil {
+		logrus.Error(err)
+		return 1
+	}
+
+	plainArgs, err = cleanInputArguments(plainArgs)
+	if err != nil {
+		logrus.Error(err)
+		return 1
+	}
+
+	parameters := parseArgs(plainArgs)
+
+	_, err = decorateType(service, parameters)
+	if err != nil {
+		logrus.Error(err)
+		return 1
+	}
+
+	err = writeYAML(service, os.Stdout)
+	if err != nil {
+		logrus.Error(err)
+		return 1
+	}
+
+	return 0
+}
+
+func writeYAML(service runtime.Object, out io.Writer) error {
+	yp := printers.YAMLPrinter{}
+	err := yp.PrintObj(service, out)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Split input parameter of type "--foo=bar" to ["--foo", "bar"]
+func splitArguments(argument string) ([2]string, error) {
+	if strings.Count(argument, "=") > 1 {
+		err := fmt.Errorf("argument %s is invalid", argument)
+		return [2]string{}, err
+	}
+	splitArgs := strings.Split(argument, "=")
+	return [2]string(splitArgs), nil
+}
+
+// Checks and fixes if input parameters are of type "--foo=bar"
+// returns fixed argument list
+func cleanInputArguments(arguments []string) ([]string, error) {
+	var fixedArguments []string
+	copy(fixedArguments, arguments)
+	for _, argument := range arguments {
+		if strings.Contains(argument, "=") {
+			fixed, err := splitArguments(argument)
+			if err != nil {
+				return nil, err
+
+			}
+			fixedArguments = append(fixedArguments, fixed[0], fixed[1])
+		} else {
+			fixedArguments = append(fixedArguments, argument)
+		}
+	}
+	return fixedArguments, nil
+}
+
+// Takes the input arguments and outputs them as separate values
+// parameterName = "Parent.child1.child2. ... .childn"
+// [parameterName-1 value-1, ...., parameterName-n value-n ] -> [[parent-1, child-1, ...child-n, value1],...]
+func parseArgs(args []string) [][]string {
+	var splitParameters [][]string
+	argsLength := len(args)
+	for index, parameterNames := range args[1:] {
+		if index%2 == 0 || index+2 == argsLength {
+			continue
+		}
+		parameterAndValue := strings.Split(strings.TrimPrefix(parameterNames, "--"), ".")
+		parameterAndValue = append(parameterAndValue, args[index+2])
+		splitParameters = append(splitParameters, parameterAndValue)
+	}
+	return splitParameters
+}
+
+// Iterates over all parameters and tries to set them
+func decorateType(serviceType runtime.Object, fieldNames [][]string) (runtime.Object, error) {
+	for _, v := range fieldNames {
+		_, err := setParameter(serviceType, v[:len(v)-1], v[len(v)-1])
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return serviceType, nil
+}
+
+// Iterates through service Type to find and set the required parameters
+// logs error and exits if parameter does not match any in Type specified field names
+func setParameter(serviceType runtime.Object, parameterHierarchy []string, value string) (runtime.Object, error) {
+	reflectedServiceType := reflect.ValueOf(serviceType).Elem()
```

Comment:
Why does this method operate on a copy of the object (I think) and not on a pointer? I think the "serviceType" argument should be a pointer and all the reflection magic should operate on that.